third_party/amd/lib/Analysis/RangeAnalysis.cpp

```diff
@@ -474,4 +474,31 @@ TritonIntegerRangeAnalysis::collectAssumptions(Operation *rootOp,
   return assumptions;
 }
 
+struct FoldTrueCmpIOp : OpRewritePattern<arith::CmpIOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  FoldTrueCmpIOp(MLIRContext *context, std::shared_ptr<DataFlowSolver> solver)
+      : OpRewritePattern(context), solver(std::move(solver)) {};
+
+  LogicalResult matchAndRewrite(arith::CmpIOp cmpOp,
+                                PatternRewriter &rewriter) const override {
+    if (cmpIIsStaticallyTrue(*solver, cmpOp)) {
+      if (failed(mlir::dataflow::maybeReplaceWithConstant(*solver, rewriter,
+                                                          cmpOp.getResult()))) {
+        LDBG("failed to replace with constant op: " << cmpOp);
```

Comment:
Should this `return failure()` otherwise we will keep rerunning it until the default max # trials (10) hit?

Reply:
done